### PR TITLE
Added configuration example in README.md when configuring a GraphServiceClient behind a corporate HTTP proxy

### PR DESCRIPTION
**Pull request by dabla**
Added configuration example in README.md when configuring a GraphServiceClient behind a corporate HTTP proxy

### CODE REVIEW

**Review by baywet — CHANGES_REQUESTED** [review]
Thank you for the contribution.
Rather than duplicating information, what would you think about simply adding a Note on this readme, with a link that points to [this page](https://learn.microsoft.com/en-us/graph/sdks/customize-client?tabs=python#configuring-the-http-proxy-for-the-client) ?

I understand the Azure Identity proxy configuration is not setup in that page, but you could PR the code snippet over there.

**Comment by dabla** [comment]
> Thank you for the contribution. Rather than duplicating information, what would you think about simply adding a Note on this readme, with a link that points to [this page](https://learn.microsoft.com/en-us/graph/sdks/customize-client?tabs=python#configuring-the-http-proxy-for-the-client) ?
> 
> I understand the Azure Identity proxy configuration is not setup in that page, but you could PR the code snippet over there.

I would like to adapt that [page](https://github.com/microsoftgraph/microsoft-graph-docs-contrib/blob/main/concepts/sdks/customize-client.md) but I can't find back the snippets in the repository.

**Comment by baywet** [comment]
here it is
https://github.com/microsoftgraph/msgraph-snippets-python/blob/main/src/snippets/custom_clients.py

**Comment by dabla** [comment]
> here it is https://github.com/microsoftgraph/msgraph-snippets-python/blob/main/src/snippets/custom_clients.py

Thank you Vincent! :)

**Comment by dabla** [comment]
I cannot apply this example in the snippets as they are using another provider which doesn't support proxies in the constructor...

https://github.com/microsoftgraph/msgraph-snippets-python/pull/23

**Comment by baywet** [comment]
Thanks for letting me know, I joined the conversation over there

**Comment by dabla** [comment]
Hello Vincent, I think we can now merge this note refering the snippets also into the README.md now that Jason has merged the snippets?

File: README.md [hunk 1]
```diff
@@ -93,14 +93,45 @@ The above example uses default scopes for [app-only access](https://learn.micros
...
+http_client = GraphClientFactory.create_with_default_middleware(
+    client=AsyncClient(proxies=httpx_proxies,
+                       timeout=Timeout(timeout=60.0)))  # HTTP timeout connection set to 60 seconds
+auth_provider = AzureIdentityAuthenticationProvider(credentials=credentials, scopes=scopes)
+client = GraphServiceClient(request_adapter=GraphRequestAdapter(auth_provider=auth_provider, client=http_client))
 ```
 
+> **Note**: Be careful as the ClientSecretCredential isn't based on the [httpx](https://www.python-httpx.org/) client but is rather using the [requests](https://requests.readthedocs.io/en/latest/) library
+underneath to make the authentication calls, so the proxies dict is a bit different for the ClientSecretCredential class as
+opposed to the [httpx](https://www.python-httpx.org/) AsyncClient.  That's why we need to adapt the proxies dict before passing it to the [httpx](https://www.python-httpx.org/) AsyncClient.
```
Comment (baywet) [inline README.md]:
```suggestion
> **Note**: Refer to the [following documentation page](https://learn.microsoft.com/graph/sdks/customize-client?tabs=python#configuring-the-http-proxy-for-the-client) if you need to configure an HTTP proxy.
```

**Review by baywet — CHANGES_REQUESTED** [review]

**Review by baywet — APPROVED** [review]
Thanks for making the changes!

**Comment by baywet** [comment]
@dabla would you mind replicating the changes [on the beta repo](https://github.com/microsoftgraph/msgraph-beta-sdk-python) please?

**Comment by dabla** [comment]
> baywet

Just created a pull request for it: https://github.com/microsoftgraph/msgraph-beta-sdk-python/pull/299

**Comment by dabla** [comment]
> Thanks for making the changes!

Thank you for guiding me through this process!

**Comment by baywet** [comment]
no worries, thanks for the contributions!

**Pull request merged**